tempesta_fw/http.c

```diff
@@ -3930,7 +3931,9 @@ tfw_http_msg_process(void *conn, TfwFsmData *data)
 		if (likely(r == T_OK || r == T_POSTPONE)) {
 			data->skb->next = data->skb->prev = NULL;
 			data->trail = !next ? trail : 0;
-			r = __tfw_http_msg_process(conn, data);
+			r = TFW_CONN_TLS((TfwConn *)conn) && tfw_http2_context(conn)
```

Comment:
Seems like `TFW_CONN_TLS((TfwConn *)conn)` is extra here. HTTP2 recommends using HTTP2 over TLS (`h2`) but it still allows HTTP2 over TCP (`h2c`). Personally I don't understand why the unencrypted connections are allowed by standard, but we have to follow them.

Reply:
It has been decided not to support non-TLS HTTP/2 in TempestaFW: standard allows HTTP/2 over plain TCP, but does not force to use it.